Add SCE contents (#36)
AlexsLemonade/scpca-docs: docs/sce_file_contents.md

| `alevinfry_version` | Version of `alevin-fry` used for mapping and quantification                                                                    |
| `af_permit_type`    | `alevin-fry generate-permit-list` method used for filtering cell barcodes                                                      |
| `af_resolution`     | `alevin-fry quant` resolution mode used                                                                                        |
| `usa_mode`          | Boolean indicating whether quantification was done using `alevin-fry` USA mode                                                 |
| `af_num_cells`      | Number of cells reported by `alevin-fry`                                                                                       |
| `transcript_type`   | Transcripts included in gene counts: `spliced` for single-cell samples and `unspliced` for single-nuclei                       |
| `miQC_model`        | The model object that `miQC` fit to the data and was used to calculate `prob_compromised`. Only present for `filtered` objects |

> ## Additional SingleCellExperiment components for CITE-seq samples

Reviewer: Do we want to put write SingleCellExperiment in code format as you are doing with the first header? I don't particularly have a preference, but we should be consistent on which way we choose to do it.

Author: I was going back and forth and forgot I did use code format before. I'll use that everywhere.

# Single-cell gene expression file contents

Single cell or single-nuclei gene expression data (filtered or unfiltered) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).
This object contains the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like CITE-seq, data from additional cell-based assays.

We highly encourage you to familiarize yourself with the general object structure and functions available as part of the [`SingleCellExperiment` package](https://bioconductor.org/packages/3.13/bioc/html/SingleCellExperiment.html) from Bioconductor.
Below we present some details about the specific contents of the objects we provide.

To begin, you will need to load the `SingleCellExperiment` package and read the RDS file:

```r
library(SingleCellExperiment)
sce <- readRDS("SCPCL000000_filtered.rds")
```

## Common fields

### Expression counts

> For both unfiltered and filtered single-cell and single-nucleotide experiments, the primary cell by gene RNA-seq expression count data is provided as the `"counts"` assay.

Reviewer: ```suggestion
The `"counts"` assay of the `SingleCellExperiment` object for single-cell and single-nucleus experiments (both unfiltered and filtered) contains the primary RNA-seq expression data.
```
Something about this sentence wasn't sitting right with me (maybe partially because you accidentally said single-nucleotide 😆), but I also think you want to emphasize that the gene expression data is found in the `counts` assay, and less about the structure since the next few sentences explain that. So I reordered it and tried to get rid of some of the repetitiveness found in the next sentences.

Reviewer: Oh you are right I just looked through some of the other docs and we are using single-nuclei. I think we should stick with that as I'm pretty sure that's also how 10X references it.

> Each column represents a cell or droplet, each row a gene.
> Column names are cell barcode sequences and row names are Ensembl gene IDs. 
> This data is stored as a sparse matrix that can be accessed with the following R code:

Reviewer: ```suggestion
The data is stored as a sparse matrix, and each column represents a cell or droplet, each row a gene.
Column names are cell barcode sequences and row names are Ensembl gene IDs. 
The `counts` assay can be accessed with the following R code:
```
I think we should first say this is a matrix and then describe the contents of the matrix.

| `subsets_mito_percent`  | Percent of all UMI counts assigned to mitochondrial genes                         |
| `total`                 | Total UMI count for RNA-seq data and any alternative experiments (i.e., CITE-seq) |

### Gene information and metrics

Gene information and metrics calculated from the RNA-seq expression data are stored as a `DataFrame` in the `rowData` slot, with the Ensembl ID as the names of the rows.

```r
gene_info <- rowData(sce)
```

The following columns are included for all genes. 
Metrics were calculated using the [`scuttle::addPerFeatureQCMetrics`](https://rdrr.io/github/LTLA/scuttle/man/addPerFeatureQCMetrics.html) function.

| Column name   | Contents                                                         |
| ------------- | ---------------------------------------------------------------- |
| `gene_symbol` | [HUGO](https://www.genenames.org) gene symbol, if defined        |
| `mean`        | Mean count across all cells/droplets                             |
| `detected`    | Number of cells in which the gene was detected (gene count > 0 ) |

### Experiment metadata

> Metadata associated with data processing is included in the `metadata` slot as a list.

Reviewer: Since most of these fields reference how the sample was processed should we add in a link back to the processing information here for more information on what each of these fields/ alevin-fry parameters actually mean?

| `mapped_reads`      | Number of reads successfully mapped                                                                      |
| `mapping_tool`      | Pipeline used for mapping and quantification (`alevin-fry` for all current data in ScPCA)                |
| `alevinfry_version` | Version of `alevin-fry` used for mapping and quantification                                              |
| `af_permit_type`    | `alevin-fry generate-permit-list` method used for filtering cell barcodes                                |
| `af_resolution`     | `alevin-fry quant` resolution mode used                                                                  |
| `usa_mode`          | Boolean indicating whether quantification was done using `alevin-fry` USA mode                           |
| `af_num_cells`      | Number of cells reported by `alevin-fry`                                                                 |
| `transcript_type`   | Transcripts included in gene counts: `spliced` for single-cell samples and `unspliced` for single-nuclei |

## Additional fields for filtered cells

> After {ref}`filtering for empty droplets <processing_information:Filtering cells>`, some additional fields are added to the `SingleCellExperiment` object.

Reviewer: ```suggestion
After {ref}`filtering and removing empty droplets <processing_information:Filtering cells>`, some additional fields are added to the `SingleCellExperiment` object.
```
I don't know if this is the best way to fix this but `filtering for empty droplets` made it sound like we were selecting for those to me, so I'm trying this to see if it helps clarify it.

Within `colData`, we add the column `prob_compromised`, which denotes the probability that a cell is compromised (i.e., dead or damaged), as calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html). 
We also add to the `metadata` a `miQC_model` item, which includes the model that `miQC` fit to the data and which was used to calculate `prob_compromised`. 

## Additional fields for CITE-seq samples

CITE-seq data, when present, is included within the `SingleCellExperiment` as an "Alternative Experiment" named `"CITEseq"` , which can be accessed with the following command:

```r
altExp(sce, "CITEseq")
```

> Within this, the main data matrix is again found in the `counts` assay, with each column corresponding to a cell or droplet (in the same order as the parent `SingleCellExperiment`) and each row corresponding to an antibody derived tag (ADT).

Reviewer: ```suggestion
Within this, the main expression matrix is again found in the `counts` assay, with each column corresponding to a cell or droplet (in the same order as the parent `SingleCellExperiment`) and each row corresponding to an antibody derived tag (ADT).

To access the counts matrix, specifically for CITE-seq data, you can use the following command: 

```r
cite_counts <- counts(altExp(sce, "CITEseq"))
```
I think it might be helpful to show here how you would do the same things in parallel to what we showed with the RNA-seq data. I know its repetitive but it could be helpful to people who don't have as much experience with working with multi-modal data in particular.

Author: Agree with the idea here, but I don't think of this as expression? I guess protein expression? I'll probably use expression unless I can come up with something better.
